### PR TITLE
authz: use partial perms for initial sync

### PR DESCRIPTION
**Pull request by unknwon**
When do the initial sync of user and repo permissions, use the partial results returned from authz provider, so we don't waste expansive API calls when the user/repo is urgent to have something to show.

Fixes https://github.com/sourcegraph/sourcegraph/issues/9157

### CODE REVIEW

File: enterprise/cmd/repo-updater/authz/perms_syncer.go [hunk 1]
```diff
@@ -488,16 +500,24 @@ type schedule struct {
 
 // scheduledUser contains information for scheduling a user.
 type scheduledUser struct {
-	Priority
-	UserID     int32
-	NextSyncAt time.Time
+	priority   Priority
+	userID     int32
+	nextSyncAt time.Time
```
Comment (keegancsmith) [inline enterprise/cmd/repo-updater/authz/perms_syncer.go]: meta: including refactorings like this make it much harder to review this PR. Rather separate those out into a different commit or PR please.

File: enterprise/cmd/repo-updater/authz/perms_syncer.go [hunk 2]
```diff
@@ -247,7 +253,7 @@ func (s *PermsSyncer) syncUserPerms(ctx context.Context, userID int32) (err erro
 // syncRepoPerms processes permissions syncing request in repository-centric way.
 // It discards requests that are made for non-private repositories based on the
 // value of "repo.private" column.
-func (s *PermsSyncer) syncRepoPerms(ctx context.Context, repoID api.RepoID) (err error) {
+func (s *PermsSyncer) syncRepoPerms(ctx context.Context, repoID api.RepoID, noPerms bool) (err error) {
```
Comment (keegancsmith) [inline enterprise/cmd/repo-updater/authz/perms_syncer.go]: nit: document `noPerms`

**Comment by keegancsmith** [comment]
Do we have any concerns about bug reports from admins about partial/no repos being visible? Then the only clue is a debug log. This might lead to very confusing behaviour. Is it worth doing partial perms?

File: enterprise/cmd/repo-updater/authz/perms_syncer.go (hunk 1 above)
Comment (unknwon) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go]: Ah, thought about it and forgot 😞 sorry!

**Comment by unknwon** [comment]
@keegancsmith 
> Do we have any concerns about bug reports from admins about partial/no repos being visible? Then the only clue is a debug log. This might lead to very confusing behaviour. Is it worth doing partial perms?

I agree it is confusing to debug with partial perms, but the fact with repo-centric syncing is that until we finish syncing perms for all repos, permissions are always "partial" (i.e. we don't know if the user has access to remaining repos in the queue). 

This is not a problem if we only do user-centric syncing (what we do with on-demand syncing).
